c/certificateService.c

```diff
@@ -118,36 +119,83 @@ static void respondWithBadRequest(HttpResponse *response) {
 
     jsonStart(p);
     {
-      jsonAddString(p, "error", "The length of the certificate is longer than 4096 bytes");
+      jsonAddString(p, "error", value);
     }
     jsonEnd(p);
 
     finishResponse(response);
 }
 
+
 static int serveMappingService(HttpService *service, HttpResponse *response)
 {
   HttpRequest *request = response->request;
 
-  if (!strcmp(request->method, methodPOST))
-  {
-    RUsermapParamList *userMapCertificateStructure 
-      = (RUsermapParamList*)safeMalloc31(sizeof(RUsermapParamList),"RUsermapParamList");
-    memset(userMapCertificateStructure, 0, sizeof(RUsermapParamList));
+  if (!strcmp(request->method, methodPOST)) {
+    RUsermapParamList userMapStructure = {0};
 
-    if(request->contentLength > sizeof(userMapCertificateStructure->certificate) || request->contentLength < 0) {
-      respondWithBadRequest(response);
+    int urlLength = strlen(request->uri);
+    if (urlLength < 0 || urlLength > 64) {
+      respondWithBadRequest(response, "URL length is not in range from 0 to 64 bytes.");
       return 0;
     }
+    char translatedURL[urlLength + 1];
+    memcpy(translatedURL, request->uri, strlen(request->uri));
+    a2e(translatedURL, sizeof(translatedURL));
+    char *found = strstr(translatedURL,"x509");
+
+    if (found != NULL) {
+    //  Certificate to user mapping
+      if (request->contentLength > sizeof(userMapStructure.certificate) || request->contentLength < 1) {
+        respondWithBadRequest(response, "The length of the certificate is longer than 4096 bytes");
+        return 0;
+      }
 
-    userMapCertificateStructure->certificateLength = request->contentLength;
-    memset(userMapCertificateStructure->certificate, 0, request->contentLength);
-    memcpy(userMapCertificateStructure->certificate, request->contentBody, request->contentLength);
+        userMapStructure.certificateLength = request->contentLength;
+        memset(userMapStructure.certificate, 0, request->contentLength);
```

Comment:
Nit: this line does nothing:
* `userMapStructure` is 0 anyway because
```c
RUsermapParamList userMapStructure = {0};
```
* `userMapStructure.certificate` is immediately overwritten by `request->contentBody` with the same length (`request->contentLength`) as in the `memset` call